src/libs/workspace-utils.ts

```diff
@@ -36,8 +36,10 @@ export const isGoogleWorkspaceInfo = (workspace: WorkspaceInfo): workspace is Go
   return workspace.cloudPlatform === 'Gcp';
 };
 
+export type WorkspaceAccessLevel = 'PROJECT_OWNER' | 'OWNER' | 'WRITER' | 'READER' | 'NO ACCESS';
```

Comment:
I don't like that we duplicate these in `utils.ts`:

export const workspaceAccessLevels = ['NO ACCESS', 'READER', 'WRITER', 'OWNER', 'PROJECT_OWNER'];

Reply:
@msilva-broad suggests (in https://docs.google.com/document/d/17jMMUyBABc6HFCsbfj6MURWHqpFxWC1f0zielaiw73s/edit):

Avoid using Typescript Enum type. Use a discriminating union of string values instead
	i.e. export type CloudPlatform = ‘GCP’ | ‘Azure’
Defining reusable constants for the enum value is also possible

	export const cloudPlatforms: Record<CloudPlatform, CloudPlatform> = {
	  GCP: ‘GCP’,
	  Azure: ‘Azure’
	}

__________________________________________________________________
Here is another approach from StackOverflow:

https://stackoverflow.com/questions/44480644/string-union-to-string-array

const ALL_SUITS = ['hearts', 'diamonds', 'spades', 'clubs'] as const;
type SuitTuple = typeof ALL_SUITS; // readonly ['hearts', 'diamonds', 'spades', 'clubs']
type Suit = SuitTuple[number];  // "hearts" | "diamonds" | "spades" | "clubs

Reply:
@nawatts extended the second example somewhat:

```
const suit = {
  HEARTS: 'hearts',
  DIAMONDS: 'diamonds',
  SPADES: 'spades',
  CLUBS: 'clubs',
} as const;

const Suits = Object.values(suit);
type Suit = (typeof Suits)[number]; // "hearts" | "diamonds" | "spades" | "clubs

const mySuit: Suit = suit.HEARTS;

Reply:
de-duplicated
I didn't do that originally, because it messed with a bunch of other areas of the project, but I realized if I left the convenience functions (`canWrite`, `canRead`, and `isOwner`) in place, then the rest was minimal.